Worker startup race condition causes 2 SessionStart hook errors on every first session

## Bug Description

On every first Claude Code session start (after a reboot or when no worker daemon is running), the claude-mem SessionStart hooks produce 2 errors. Immediately closing and reopening Claude Code (second session) produces 0 errors because the worker daemon is already running.

## Root Cause

There is a race condition between the MCP server startup and the SessionStart hooks:

1. Claude Code starts → MCP servers initialize → claude-mem MCP server (`mcp-server.cjs`) starts
2. MCP server detects worker not running → starts worker daemon on port 37777
3. SessionStart hooks run → `worker-service.cjs start` also tries to start the worker → **fails with "port 37777 in use"**
4. `worker-service.cjs hook claude-code context` may also fail because the worker is still stabilizing

## Evidence from logs

```
[09:41:45.312] [INFO ] Claude-mem search server started
[09:41:45.357] [ERROR] Worker not available {"workerUrl":"http://127.0.0.1:37777"}
[09:41:45.540] [INFO ] Starting worker daemon    ← TWO parallel starts!
[09:41:45.540] [INFO ] Starting worker daemon
[09:41:45.666] [INFO ] HTTP server started {host=127.0.0.1, port=37777, pid=1619}
[09:41:45.667] [INFO ] Worker started
[09:41:45.669] [ERROR] ✗ Worker failed to start Failed to start server. Is port 37777 in use?
```

On second session start (worker already running):
```
[09:41:56.979] [INFO ] Worker already running and healthy  ← No errors
```

## Expected Behavior

The `worker-service.cjs start` command should:
1. Check if the worker is already running/starting on the target port
2. If another process already bound the port, verify it's actually the worker (health check)
3. If health check passes, treat it as success (exit 0) instead of error (exit 1)

## Suggested Fix

In the worker start logic, when the port is already in use:
```javascript
// Instead of immediately failing:
// throw new Error("Failed to start server. Is port 37777 in use?")

// Check if the existing process IS the worker:
const isWorkerHealthy = await checkHealth("http://127.0.0.1:37777/api/health");
if (isWorkerHealthy) {
  console.log("Worker already running and healthy (started by MCP server)");
  process.exit(0);  // Success, not error
}
```

## Workaround

Added a pre-warm SessionStart hook (`claude-mem-worker-guard.sh`) that waits for the worker to become healthy before the plugin hooks run. This mitigates the issue but doesn't fix the root cause.

## Environment

- Claude Code v2.1.80
- claude-mem v10.5.5
- macOS (Apple Silicon)
- Bun 1.3.11

--- a/src/services/worker-service.ts
+++ b/src/services/worker-service.ts
@@ -1200,7 +1200,18 @@ async function main() {
       });
 
       const worker = new WorkerService();
-      worker.start().catch((error) => {
+      worker.start().catch(async (error) => {
+        // Port race: when the MCP server and SessionStart hook both spawn a daemon
+        // concurrently, one will lose the bind race with EADDRINUSE or Bun's equivalent
+        // "port in use" error. If the winner is already healthy, exit cleanly (#1447).
+        const isPortConflict = error instanceof Error && (
+          (error as NodeJS.ErrnoException).code === 'EADDRINUSE' ||
+          /port.*in use|address.*in use/i.test(error.message)
+        );
+        if (isPortConflict && await waitForHealth(port, 3000)) {
+          logger.info('SYSTEM', 'Duplicate daemon exiting — another worker already claimed port', { port });
+          process.exit(0);
+        }
         logger.failure('SYSTEM', 'Worker failed to start', {}, error as Error);
         removePidFile();
         // Exit gracefully: Windows Terminal won't keep tab open on exit 0

--- a/tests/services/worker-daemon-port-race.test.ts
+++ b/tests/services/worker-daemon-port-race.test.ts
@@ -1,0 +1,44 @@
+import { describe, it, expect } from 'bun:test';
+import { readFileSync } from 'fs';
+import { join } from 'path';
+
+/**
+ * Source-inspection tests for Issue #1447: Worker startup race condition
+ *
+ * When the MCP server and SessionStart hook both spawn a daemon concurrently,
+ * one daemon loses the port bind race (EADDRINUSE / Bun's "port in use" error).
+ * The loser should detect this, verify the winner is healthy, and exit cleanly
+ * instead of logging an ERROR that clutters the user's session start output.
+ *
+ * These are source-inspection tests because the race is non-deterministic and
+ * requires a real concurrent multi-process scenario to reproduce reliably.
+ */
+
+const WORKER_SERVICE_PATH = join(import.meta.dir, '../../src/services/worker-service.ts');
+const source = readFileSync(WORKER_SERVICE_PATH, 'utf-8');
+
+describe('Worker daemon port-race guard (#1447)', () => {
+  it('detects EADDRINUSE error code in the port-conflict check', () => {
+    expect(source).toContain("code === 'EADDRINUSE'");
+  });
+
+  it('detects Bun port-in-use message via regex in the port-conflict check', () => {
+    expect(source).toContain('/port.*in use|address.*in use/i.test(error.message)');
+  });
+
+  it('calls waitForHealth before exiting on a port conflict', () => {
+    // The guard must verify the winner is actually healthy before exiting,
+    // otherwise a non-worker process on the port would suppress a real error.
+    expect(source).toContain('isPortConflict && await waitForHealth(port,');
+  });
+
+  it('uses async catch handler to allow awaiting waitForHealth', () => {
+    // The .catch() must be async so it can await the health check.
+    expect(source).toContain('worker.start().catch(async (error) =>');
+  });
+
+  it('logs info (not error) when cleanly exiting after port race', () => {
+    // Must not call logger.failure() / logger.error() on the clean exit path.
+    expect(source).toContain("logger.info('SYSTEM', 'Duplicate daemon exiting");
+  });
+});

